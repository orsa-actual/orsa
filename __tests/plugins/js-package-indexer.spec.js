const jsPackageIndexer = require('../../src/plugins/js-package-indexer');

describe('Javascript Package Indexer', () => {
  it('should index', async () => {
    let ind = null;
    await jsPackageIndexer({}, {
      store: {
        query: () => ({
          data: {
            projects: [
              {
                id: '1',
                name: 'bar',
              },
            ],
          },
        }),
        getById: () => ({
          dependencies: [
            {
              name: 'foo',
              version: '1.0.5',
              requestedVersion: '^1.0.0',
              type: 'direct',
            },
          ],
        }),
        updateIndex: (name, index) => {
          ind = index;
        },
      },
      logger: {
        log: () => {},
        error: () => {},
      },
    });
    expect(ind).toMatchSnapshot();
  });
});
