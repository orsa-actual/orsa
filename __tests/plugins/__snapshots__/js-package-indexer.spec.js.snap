// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Javascript Package Indexer should index 1`] = `
Object {
  "foo": Object {
    "name": "foo",
    "requestedVersions": Array [
      Object {
        "name": "foo",
        "projects": Array [
          Object {
            "project": undefined,
            "type": "direct",
            "version": undefined,
          },
        ],
        "version": "^1.0.0",
      },
    ],
    "versions": Array [
      Object {
        "name": "foo",
        "projects": Array [
          Object {
            "project": undefined,
            "type": "direct",
            "version": undefined,
          },
        ],
        "version": "1.0.5",
      },
    ],
  },
}
`;
